libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -46,6 +46,24 @@ AP_AHRS_NavEKF::AP_AHRS_NavEKF(uint8_t flags) :
     _dcm_matrix.identity();
 }
 
+// init sets up INS board orientation
+void AP_AHRS_NavEKF::init()
+{
+#if !HAL_NAVEKF2_AVAILABLE && HAL_NAVEKF3_AVAILABLE
+    if (_ekf_type.get() == 2) {
```

Comment:
More magic?

A pre-arm check that your configured EKF is available will lead to less complicated debugging.

Reply:
that would lead to a lot more confusion as users of these boards wonder why their existing config no longer works, and also it makes then switching between master and stable harder